src/tabs/presets/presets.js

```diff
@@ -303,13 +306,35 @@ TABS.presets.tryLoadPresets = function() {
     });
 };
 
+TABS.presets.checkPresetSourceVersion = function() {
+    return new Promise((resolve, reject) => {
+        if (this.majorVersion === this.presetsRepo.index.majorVersion) {
+            resolve();
+        } else {
+            const versionRequired = `${this.majorVersion}.X`;
+            const versionSource = `${this.presetsRepo.index.majorVersion}.${this.presetsRepo.index.minorVersion}`;
+
+            const dialogSettings = {
+                title: i18n.getMessage("presetsWarningDialogTitle"),
+                text: i18n.getMessage("presetsVersionMismatch", {"versionRequired": versionRequired, "versionSource":versionSource}),
+                buttonYesText: i18n.getMessage("yes"),
+                buttonNoText: i18n.getMessage("no"),
+                buttonYesCallback: () => resolve(),
+                buttonNoCallback: () => reject("Prset source version mismatch"),
+            };
+
+            GUI.showYesNoDialog(dialogSettings);
+        }
+    });
+};
```

Comment:
Why this is a promise? I don'st see anything here what needs it to wrap it in the promise.

Reply:
`GUI.showYesNoDialog(dialogSettings)` is not synchronous operation and work on callbacks.
`checkPresetSourceVersion` is a function that is nice to call with .then chaininig.
So I could make this function work with callbacks, but wrapping with promises is nicer here.

Reply:
A, cool, didn't notice that bit. Thanks 😃